spring-kafka/src/main/java/org/springframework/kafka/support/DefaultKafkaHeaderMapper.java

```diff
@@ -385,18 +387,16 @@ private Object decodeValue(Header h, Class<?> type) throws IOException, LinkageE
 		return value;
 	}
 
-	@SuppressWarnings("unchecked")
-	@Nullable
 	private Map<String, String> decodeJsonTypes(Headers source) {
-		Map<String, String> types = null;
+		Map<String, String> types = Collections.emptyMap();
```

Comment:
it is common good practise to return empty colletion as opposed to null to avoid NPE as per Effective Java.